uni/ulsp/workspace.icn

```diff
@@ -627,17 +638,54 @@ class Context(uniAll, uri, line, lineNum, charNum, objectName, methodName, conte
       }
 
       #-----------------------------------------------#
-      #----------- Case "inside comment" -------------#
+      #------- Case "inside comment or string" -------#
       #-----------------------------------------------#
 
       line ? {
          single_quote := 0
          double_quote := 0
+         backslash_count := 0
          while (&pos < charNum) do {
             ch := move(1) | break
-            if ch == "\'" then single_quote +:= 1
-            if ch == "\"" then double_quote +:= 1
-            if (ch == "#") & ((single_quote % 2) = 0 & (double_quote % 2) = 0) then return "comment"
+            if ch == "\'" then {
+               temppos := &pos
+               move(-1)
+               while move(-1) == "\\" do backslash_count +:= 1
+               if (backslash_count % 2) = 0 then single_quote +:= 1
+               backslash_count := 0
+               &pos := temppos
+            }
+            if ch == "\"" then {
+               temppos := &pos
+               move(-1)
+               while move(-1) == "\\" do backslash_count +:= 1
+               if (backslash_count % 2) = 0 then double_quote +:= 1
+               backslash_count := 0
+               &pos := temppos
+            }
+            if (ch == "#") & ((single_quote % 2) = 0) & ((double_quote % 2) = 0) then return "comment"
+         }
+         if ((single_quote % 2) ~= 0 | (double_quote % 2) ~= 0) then return "string"
+      }
+
+      #-----------------------------------------------#
+      #--------- Case "package dual colon" -----------#
+      #-----------------------------------------------#
+
+      line ? {
+         tab(charNum)
+         move(-2)
+         if move() == ":" & move() == ":" then {
```

Comment:
this is the same as 
```
if  = "::" then {
```

Reply:
I guess it is a bit different. Since `=` will only move the scan position if there is a match.